scvi/dataset/dataset.py

```diff
@@ -53,10 +57,18 @@ def download_and_preprocess(self):
     def collate_fn(self, batch):
         indexes = np.array(batch)
         X = torch.FloatTensor(self.X[indexes]) if self.dense else torch.FloatTensor(self.X[indexes].toarray())
-        return X, torch.FloatTensor(self.local_means[indexes]), \
-            torch.FloatTensor(self.local_vars[indexes]), \
-            torch.LongTensor(self.batch_indices[indexes]), \
-            torch.LongTensor(self.labels[indexes])
+        if not self.got_spatial:
```

Comment:
Let's just have `if x_coord is not None:` here, and get rid of the `got_spatial` variable.